### PR TITLE
Retire some features

### PR DESCRIPTION
**Pull request by ankhyk**
- Retire commands "generate-uml"
 - Retire commands "generate-wireformat"
 - Retire commands "resolve-spec"
 - Retire Autorest plugin

### CODE REVIEW

File: lib/validate.ts [hunk 1]
```diff
@@ -136,7 +136,7 @@ const prettyPrintInfo = <T>(
 
 export const validateSpec = async (specPath: string, options: Options | undefined) =>
   validate(options, async (o) => {
-    const validator = new SemanticValidator(specPath, null, o);
+    const validator = new SemanticValidator(specPath, null);
```
Comment (raych1) [inline lib/validate.ts:95]: @jianyexi , do unified pipeline and static validation pass in the option?

Comment (jianyexi) [reply lib/validate.ts:95]: It just uses the 'validateSpec'

**Comment by leniatgh** [comment]
@keryul there's conflict, please resolve. BTW, how did you find the unused dependencies?

**Comment by ankhyk** [comment]
> @keryul there's conflict, please resolve. BTW, how did you find the unused dependencies?

@leni-msft I list all the dependencies the retired features imported, and check if they are used in other files.
If no other file imports the dependency, I determine that it is an unused dependency.

**Comment by leniatgh** [comment]
> > @keryul there's conflict, please resolve. BTW, how did you find the unused dependencies?
> 
> @leni-msft I list all the dependencies the retired features imported, and check if they are used in other files. If no other file imports the dependency, I determine that it is an unused dependency.

I see, please use the [depcheck tool](https://www.npmjs.com/package/depcheck) to check again

**Comment by changlong-liu** [comment]
Hi @raych1, @leni-msft , 
with some local try with the PR branch, I think the testmodeler and mock-service-host are good with this change.

**Comment by ankhyk** [comment]
> > > @keryul there's conflict, please resolve. BTW, how did you find the unused dependencies?
> > 
> > 
> > @leni-msft I list all the dependencies the retired features imported, and check if they are used in other files. If no other file imports the dependency, I determine that it is an unused dependency.
> 
> I see, please use the [depcheck tool](https://www.npmjs.com/package/depcheck) to check again

Great tool!
I checked with depcheck and there are no unused dependencies in this branch

File: ChangeLog.md [hunk 2]
```diff
@@ -2,6 +2,11 @@
 
 ## 09/19/2022 3.1.0
 
+- Retire some features
+  - Retire commands "generate-uml"
+  - Retire commands "generate-wireformat"
+  - Retire commands "resolve-spec"
+  - Retire Autorest plugin
```
Comment (leniatgh) [inline ChangeLog.md:9]: need use a new minor version number

**Review by leniatgh — APPROVED** [review]

File: ChangeLog.md (hunk 2 above)
Comment (raych1) [reply ChangeLog.md:9]: @keryul , need to bump minor version as 3.2.0

**Review by raych1 — APPROVED** [review]

**Pull request merged**